shapeshift/web
# feat: gracefully handle accounts change

## Description

This PRs adds support for gracefully handling accounts change from MetaMask-like wallets.

## Notice




## Pull Request Type


## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/2579


## Risk

High, needs a full wallet regression testing and ensuring there are no unnecessary reloads, both on the current flow and while switching accounts.

## Testing


### Engineering



- Connect MM or XDEFI with an account
- Switch the account in your wallet
- The app should refresh to reflect the change
- This should work both on wallet connection through the wallet connect modal as well as on app reload
- Ensure there are no console errors when switching accounts

### Operations

- See engineering testing notes
- Ensure the app doesn't refresh on incoming Txs
- Ensure wallet connection, disconnection still works
- Ensure there are no regressions when switching from MetaMask/XDEFI/TallyHo, to another wallet and back

Note: While testing against prod, you might notice inconsistencies in the current disconnect behavior. That's expected, the guts of these PRs is that they fix an issue we currently have, where accounts/network changed are only handled if you connect through the wallet connect menu, but if you connect to the app by reloading the page, they are never handled and produce no effect at all currently.
Please make sure to test against the disconnect behavior in prod by explicitly connecting a wallet vs. reloading the app with it already connected.

## Screenshots (if applicable)
